lib/livebook/notebook/explore/intro_to_nx.livemd

```diff
@@ -1,3 +1,742 @@
+libraries that support those tensors. Nx has three primary capabilities:
+
+* In Nx, tensors backed by other platforms hold typed data in multiple,
+  named dimensions.
+* Numerical definitions, known as `defn`, support custom code with
+  tensor-aware operators and functions.
+* [Automatic differentiation](https://arxiv.org/abs/1502.05767), also known as
+  autograd or autodiff, supports common computational scenarios
+  such as machine learning, simulations, curve fitting, and probabilistic models.
+
+Here's more about each of those capabilities. Nx [tensors]() can hold
+unsigned integers (u8, u16, u32, u64),
+signed integers (s8, s16, s32, s64),
+floats (f32, f64) and brain floats (bf16).
+Tensors support backends implemented outside of Elixir, including Google's
+Excelerated Linear Algebra (XLA) and LibTorch.
+
+Numerical definitions have compiler support to allow just-in-time compilation
+that support specialized processors to speed up numeric computation including
+TPUs and GPUs.
+
+If you want to know Nx, get to know tensors first. This rapid overview will touch
+on the major libraries. Then, future notebooks will take a deep dive into working
+with tensors in detail, autograd, and backends. Then, you can dive into specific
+problem spaces like Axon, the machine learning library.
+
+## Nx and tensors
+
+Systems of equations are a central theme in numerical computing.
+These equations are often expressed and solved with multidimensional
+arrays. For example, this is a two dimensional array:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+In Elixir, you would express a similar data structure using
+a list of lists, like this:
+
+```elixir
+[
+  [1, 2],
+  [3, 4]
+]
+```
+
+This data structure works fine within many functional programming
+algorithms, but breaks down with deep nesting and random access.
+
+On top of that, Elixir numeric types lack optimization for many numerical
+applications. They work fine when programs
+need hundreds or even thousands of calculations. They tend to break
+down with traditional STEM applications when a typical problem
+needs millions of calculations.
+
+In Nx, you'll express multi-dimensional data using typed tensors. Simply put,
+a tensor is a multi-dimensional array with a predetermined shape and
+type. To interact with them, Nx relies on tensor-aware operators rather
+than `Enum.map/2` and `Enum.reduce/3`.
+
+In this section, you'll look at some of the various tools for
+creating and interacting with tensors.
+
+<!-- livebook:{"break_markdown":true} -->
+
+To get started, you'll first need to include the Nx dependency.
+In Livebook, we can simply use Mix install. At the time of this writing,
+there's no Hex package for Nx, which means you need to use a git
+dependency, like this:
+
+```elixir
+Mix.install([
+  {:exla, "~> 0.1.0-dev", github: "elixir-nx/nx", sparse: "exla"},
```

Comment:
Until we have proper support for precompiled EXLA I think we should skip this (and the EXLA section can be more theoretic, perhaps with markdown Elixir code snippets instead of actual cells). Given it's an introduction, the user will definitely not have EXLA compiled and satisfying all the dependencies and hours of compilation will get in the way of playing with the notebook :)

Reply:
I also heard you re. voice and switched most of it to we, except the areas where we're addressing the reader directly, such as exercises. Thanks for a nice edit. I think we're waiting on @josevalim to resolve the last few issues: backend vs compiler support wording, and whether we'd like a couple more interactive livebooks on a linear regression, and nx fundamentals. I will be doing a chunk of this stuff anyway for my writing, so it's no problem either way.

Reply:
Sorry for the delay. My suggestion for now is to not use EXLA, because it will take a while to compile. More specifically:

1. let's not mention backends. there is only one for now and it is quite incomplete

2. let's mention defn compilers when we talk about defn, talk how they can be optimized, and point to the EXLA folder inside the Nx project